### PR TITLE
Header wishlist integration

### PR DESCRIPTION
**Pull request by iryb**
Task https://app.asana.com/0/1202525679663009/1202720801120417

## One model in wishlist:
wishlist-header

## Empty wishlist
wishlist-0-header

### CODE REVIEW

**Comment by oleksandr-danylchenko** [comment]
We mustn't show empty lists like that: 
image
It should be either emptiness or an explicit message like "Nothing is in wishlist". I would prefer to the latter

**Comment by oleksandr-danylchenko** [comment]
Please choose the branch from which you started the current one as the merge target. In this way, the change diff will be much smaller

File: frontend/src/components/header/header.tsx [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 import { Navigate } from './navigate/navigate';
 import { PrivateElements } from './private-elements/private-elements';
 import { PublicElements } from './public-elements/public-elements';
 
 export const Header: React.FC = () => {
+  const { data: wishlist } = useGetWishlistsQuery();
+
   const user = {
-    favorites: 3,
+    favorites: wishlist?.count as number | 0,
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/header/header.tsx]: `|` here is not what you expect :sweat_smile: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_OR

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

**Comment by iryb** [comment]
> We mustn't show empty lists like that: image It should be either emptiness or an explicit message like "Nothing is in wishlist". I would prefer to the latter

### If nothing is in favorites:
image

File: backend/src/services/preferences/wishlist.service.ts [hunk 2]
```diff
@@ -253,9 +253,13 @@ const getWishlistByUserId = async (
...
   );
 
+  const wishlistCount =
+    sanitizedModelsWishlist.length + sanitizedComplectationsWishlist.length;
+
   const wishlists: WishlistsResponseDto = {
     models: sanitizedModelsWishlist,
     complectations: sanitizedComplectationsWishlist,
+    count: wishlistCount,
   };
```
Comment (oleksandr-danylchenko) [inline backend/src/services/preferences/wishlist.service.ts]: I would put the counting login on the client side. Then we'll be able to update the liked count right after the like

File: frontend/src/components/header/header.tsx [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 import { Navigate } from './navigate/navigate';
 import { PrivateElements } from './private-elements/private-elements';
 import { PublicElements } from './public-elements/public-elements';
 
 export const Header: React.FC = () => {
+  const { data: wishlist = 0 } = useGetWishlistsQuery();
+
   const user = {
-    favorites: 3,
+    favorites: wishlist && wishlist.count,
```
Comment (oleksandr-danylchenko) [inline frontend/src/components/header/header.tsx]: We shouldn't use `smth &&` check in the modern JS. There a optional [chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) in JS to work with possibly `undefined` objects

File: shared/src/common/types/preferences/wishlist-response-dto.type.ts [hunk 4]
```diff
@@ -60,6 +60,7 @@ type ComplectationResponseDto = {
 interface WishlistsResponseDto {
   models?: ModelResponseDto[];
   complectations?: ComplectationResponseDto[];
+  count: number;
```
Comment (oleksandr-danylchenko) [inline shared/src/common/types/preferences/wishlist-response-dto.type.ts]: `count` would better live on the client

**Review by oleksandr-danylchenko — CHANGES_REQUESTED** [review]

**Review by oleksandr-danylchenko — APPROVED** [review]

**Pull request merged**